scalding-core/src/test/scala/com/twitter/scalding/ExecutionTest.scala

```diff
@@ -212,6 +214,131 @@ class ExecutionTest extends WordSpec with Matchers {
         .shouldSucceed() shouldBe (1, "1")
     }
 
+    "If one write fails, the other gets cancelled" in {
+      @volatile var cancelledEx: Option[Throwable] = None
+
+      val failedTp: TypedPipe[Int] = TypedPipe.from(Seq(0)).groupAll.sum.values.map { _ => throw new Exception("oh no") }
+      val failedEx: Execution[Iterable[Int]] = failedTp.toIterableExecution
+
+      val mapCountDownLatch = new CountDownLatch(1)
+
+      val blockingTp: TypedPipe[Int] = TypedPipe.from(Seq(1)).groupAll.sum.values.map { i =>
+        // block until we are done
+        mapCountDownLatch.await()
+        i
+      }
+
+      val onCompleteCountDownLatch = new CountDownLatch(1)
+      val otherEx: Execution[Iterable[Int]] = blockingTp.toIterableExecution.onComplete { t =>
+        if (t.isFailure) {
+          // capture the exception
+          cancelledEx = t.failed.toOption
+        }
+        onCompleteCountDownLatch.countDown()
+      }
+
+      val zipped = failedEx.zip(otherEx)
+
+      zipped.shouldFail()
+
+      // wait for onComplete to finish
+      onCompleteCountDownLatch.await()
+
+      // execution should be cancelled and the flow stopped
+      assert(cancelledEx.get.isInstanceOf[FlowStopException])
+
+      // finish counting down on the map to release the thread
+      mapCountDownLatch.countDown()
+
+      // do the same on the other side
+      @volatile var cancelledEx2: Option[Throwable] = None
+
+      val failedTp2: TypedPipe[Int] = TypedPipe.from(Seq(0)).groupAll.sum.values.map { _ => throw new Exception("oh no") }
+      val failedEx2: Execution[Iterable[Int]] = failedTp2.toIterableExecution
+
+      val mapCountDownLatch2 = new CountDownLatch(1)
+
+      val blockingTp2: TypedPipe[Int] = TypedPipe.from(Seq(1)).groupAll.sum.values.map { i =>
+        // block until we are done
+        mapCountDownLatch2.await()
+        i
+      }
+
+      val onCompleteCountDownLatch2 = new CountDownLatch(1)
+      val otherEx2: Execution[Iterable[Int]] = blockingTp2.toIterableExecution.onComplete { t =>
```

Comment:
can you also make a copy of this test that instead of using `.onComplete` uses `.liftToTry.map`. This has different semantics slightly from onComplete, but should still pass in my opinion.

Reply:
i think there is a problem here where `liftToTry` uses `recoverWith` to construct the failure case, but we skip the `recoverWith` function when an execution is cancelled. as a result, we end up throwing instead of returning a failed Try.

do you think it is worth implementing a `LiftToTry` execution type, or do you have other suggestions?

Reply:
Ahh, yeah. I think just remove the test then. 

If you want to test more cases, you might consider exercising `Execution.flatMap` to make sure cancelation propagates inside if a job is started by the flatMap function.

Like if we had `Exection.unit.flatMap { _ => foo.toIterableExecution.map(waitingFn) }` we should expect that cancellation still works.